packages/core/lib/commands/test/run.js

```diff
@@ -3,13 +3,22 @@ const parseCommandLineFlags = options => {
   const grep = options.grep || options.g;
   const bail = options.bail || options.b;
   const reporter = options.reporter || options.r;
-  return {
-    mocha: {
-      grep,
-      bail,
-      reporter
-    }
-  };
+  if (reporter === undefined) {
```

Comment:
Is there some reason this conditional is needed, to exclude `reporter` when it's undefined?  Does something go wrong if you go with the simpler `return { mocha: { grep, bail, reporter } }` regardless of whether `reporter` is defined or not?

Reply:
@eggplantzzz and I found out that when the reporter is undefined, then it doesn't fall back to the truffle-config.js specification of the mocha reporter defined there. It just displays the default mocha reporter "spec". So, it needs to be excluded when the reporter is undefined.

Reply:
Oh, weird!  It might be worth adding a code comment to note that, because this looks like the sort of thing a well-meaning simplifier might accidentally remove otherwise.  Thanks for explaining that!

Reply:
Sure, I will add a comment above the code for better understanding of the condition.

Reply:
@haltman-at if you explicitly set `reporter` to the value `undefined`, it does a merge later that cycles through the keys overwriting other values that might have been set. In this case, it will find that `reporter` has a value, namely `undefined`, and overwrite the other value.

Reply:
Yes, that would make sense given what @sukanyaparashar said!